File: 250818/줄 세우기/line-up-students.py
# 키가 더 큰 항생이 앞에
# 키가 동일하면 몸무가게 더 큰 학생
# 키 == 몸무게면 번호가 더 작은 학생.

class Students :
    def __init__ (self, height, weight, number) :
        self.height = height;
        self.weight = weight;
        self.number = number;

N = int(input());

students = [];
number = 1;
for i in range(N) :
    height, weight = map(int, input().split());
    students.append(Students(height,weight,number));
    number += 1;

students.sort(key = lambda x : (-x.height, -x.weight, x.number));

for elem in students :
    print(elem.height, elem.weight, elem.number);
    

File: 250818/줄 세우기/line-up-students.java
import java.util.*;

public class Main {

    static class Students implements Comparable<Students>{
        int height;
        int weight;
        int number;

        public Students(int height, int weight, int number){
            this.height = height;
            this.weight = weight;
            this.number = number;
        }


        @Override
        public int compareTo(Students students){
            if (this.height == students.height) {

                if(this.weight == students.weight){
                    return this.number - students.number;
                }

                return students.weight - this.weight;
            }
            return students.height - this.height;
        }



    }

    public static void main(String[] args) {
        // Please write your code here.

        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();

        Students student [] = new Students[N];

        int number = 1;
        for(int i = 0; i < N; i++){
            int height = sc.nextInt();
            int weight = sc.nextInt();

            student[i] = new Students(height, weight, number++);
        }

        Arrays.sort(student);

        for (int i = 0; i < N; i++){
            System.out.println(student[i].height +" " + student[i].weight + " " + student[i].number);
        }

    }
}